File: app/src/main/java/au/com/dmg/fusioncloud/android/demo/GlobalClass.java
package au.com.dmg.fusioncloud.android.demo;

import android.app.Application;
// TODO: Will utilise this for Preauthorisation and Completion
public class GlobalClass extends Application {

    private static GlobalClass instance;

    public static GlobalClass getInstance() {
        return instance;
    }

    public static GlobalClass getContext(){
        return instance;
        // or return instance.getApplicationContext();
    }

    @Override
    public void onCreate() {
        instance = this;
        super.onCreate();
    }

}

File: app/src/main/java/au/com/dmg/fusioncloud/android/demo/data/GlobalClass.kt
package au.com.dmg.fusioncloud.android.demo.data

import android.app.Application

// TODO: Will utilise this for Preauthorisation and Completion
class GlobalClass : Application() {
    override fun onCreate() {
        context = this
        super.onCreate()
    }

    companion object {
        // or return instance.getApplicationContext();
        var context: GlobalClass? = null
            private set
            get() = field
        // or return instance.getApplicationContext();
    }
}